src/views/templates/details/tabs/scheduling/components/AffinityRulesModal.tsx

```diff
@@ -91,14 +92,15 @@ const AffinityRulesModal: React.FC<AffinityModalProps> = ({
 
   const updatedTemplate = React.useMemo(() => {
     return produce<V1Template>(template, (templateDraft: V1Template) => {
+      const draftVM = getTemplateVirtualMachineObject(templateDraft);
       if (!getAffinity(templateDraft)) {
-        templateDraft.objects[0].spec.template.spec.affinity = [];
```

Comment:
Now it's explicit that we are using a vm and not a random k8sCommon object so typescript triggered about affinity. Affinity property is not an array so I defaulted to null here